src/lib/functions/synchronous.js

```diff
@@ -36,14 +38,34 @@ const handleSynchronousFunction = function (err, result, response) {
   response.end()
 }
 
-const formatLambdaLocalError = (err) => `${err.errorType}: ${err.errorMessage}\n  ${err.stackTrace.join('\n  ')}`
+const formatLambdaLocalError = (err, acceptsHtml) =>
+  acceptsHtml
+    ? JSON.stringify({
+        errorType: err.errorType,
+        errorMessage: err.errorMessage,
+        trace: err.stackTrace,
+      })
+    : `${err.errorType}: ${err.errorMessage}\n ${err.stackTrace.join('\n ')}`
 
-const handleErr = function (err, response) {
+const renderErrorTemplate = (errString) => {
+  const regexPattern = /<!--@ERROR-DETAILS-->/g
+  const errorTemplateFile = readFileSync(join(__dirname, './templates/function-error.html'), 'utf-8')
```

Comment:
We're still reading the file each time we need to render a new error page, right? I think we could have `errorTemplateFile` in the global scope of this file, and then whenever you need to render a template you do something like:

```js
errorTemplateFile = errorTemplateFile || readFile(...)
```

So if the file has already been read, use the in-memory value. If not, read it and assign it for future invocations. Does that make sense?

Reply:
Yes and done.